regexec.c

```diff
@@ -228,8 +227,9 @@ STATIC CHECKPOINT
 S_regcppush(pTHX_ const regexp *rex, I32 parenfloor, U32 maxopenparen _pDEPTH)
 {
     const int retval = PL_savestack_ix;
-    const int paren_elems_to_push =
-                (maxopenparen - parenfloor) * REGCP_PAREN_ELEMS;
+    const SSize_t paren_bytes_to_push = sizeof(*rex->offs) * (maxopenparen - parenfloor);
+    /* Number of savestack[] entries to be filled by the paren data */
+    const int paren_elems_to_push = (paren_bytes_to_push + sizeof(*PL_savestack) - 1) / sizeof(*PL_savestack);
```

Comment:
I think the preceding comment here should note that this is rounding in disguise would be in order. I had to stare at this a while to understand what it was doing.

Reply:
I see ok I'll try to add more detailed comments as soon as I have some free time.

Reply:
Thanks. For what its worth I really appreciate you writing this patch, its a great idea.